text/0000-rfc-template.md

```diff
@@ -0,0 +1,155 @@
+
+I believe the level of granularity here matches this proposal, with some different choices which are due to cultural and technical context.
+
+### F#
+
+The F# RFC process is similar to Rust's RFC process and has a similar [template](https://github.com/fsharp/fslang-design/blob/main/RFC_template.md). It's sections are:
+
+* summary,
+* motivation,
+* detailed design,
+* drawbacks,
+* alternatives,
+* compatibility,
+* pragmatics (diagnostics, tooling, performance, scaling, culture-aware formatting/parsing),
+* unresolved questions.
+
+The first few sections and the last are similar to in this proposal. They have separate 'drawbacks' and 'alternatives' sections. In Rust, 'compatibility' is usually included in the 'detailed design' sections, and this seems to work well for us. The 'pragmatics' section is an interesting factoring and is worth considering for future work. I think it will take some work to choose appropriate sub-sections for Rust because our RFCs have broader use than in F#.
+
+### Ember
+
+Ember's RFC process was strongly influenced by Rust's and there has been much cross-pollination over the years. Their [template](https://github.com/emberjs/rfcs/blob/master/0000-template.md) is similar to earlier iterations of Rust's template.
+
+## History of the RFC template
+
+The RFC template has always been part of the RFC process but has changed over time. The major changes (and the RFC PR which proposed those change) are:
+
+* #6: initial template (though actually it existed in the repo before the PR because this was the wild-west age of the RFCs repo),
+* #32: adds a 'drawbacks' section,
+* #1636: adds a "How do we teach this?" section; the motivation was primarily to encourage authors to think about documentation,
+* #2059: changes the 'detailed design' and 'how do we teach this' sections into 'guide-level explanation' and 'reference-level explanation' sections; expands 'alternatives' to 'rationale and alternatives'. The motivation was to make the process more accessible, and because there was a feeling that the 'how do we teach this' section was not having the desired effect, in part because it was interpreted as describing the mechanism for teaching, rather than the details of describing the feature to users who have not encountered it before.
+* #2333: adds a 'prior art' section,
+* #2561: adds a 'future possibilities' section.
+
+My impression is that the guide-level/reference-level split has not worked out for us. A;though it has had some benefits in thinking about the way new features are documented, these are outweighed by the drawbacks (discussed above). Adding sections for drawbacks, alternatives, and rationale have all been beneficial, but with the benefit of hindsight, these would be better as one section rather than three. Although no one change is responsible, we can see that the effect of the evolution over time has led to the RFC template having many fine-grained sections.
```

Comment:
```suggestion
My impression is that the guide-level/reference-level split has not worked out for us. Although it has had some benefits in thinking about the way new features are documented, these are outweighed by the drawbacks (discussed above). Adding sections for drawbacks, alternatives, and rationale have all been beneficial, but with the benefit of hindsight, these would be better as one section rather than three. Although no one change is responsible, we can see that the effect of the evolution over time has led to the RFC template having many fine-grained sections.
```